### ISSUE TITLE
Cannot use module with dynamic import since Nuxt 3.4.2

### ISSUE THREAD

**Issue by WolfgangDrescher** [nuxt/nuxt#21578]
### Environment

- Operating System: Linux
- Node Version:     v16.14.2
- Nuxt Version:     3.4.2
- Nitro Version:    2.4.1
- Package Manager:  npm@9.4.2
- Builder:          vite
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -

### Reproduction

https://stackblitz.com/edit/github-ampofb?file=app.vue

In this demo Nuxt is locked to `v3.4.2`. Set the version to `3.4.1` and run `npm i && npm run dev` to see what the output should actually look like.

### Describe the bug

When I import the npm package `verovio` since nuxt 4.3.2 I will get an error `Failed to resolve entry for package`. Probably this is because Vite was bumped to `4.3` in nuxt `4.3.2`. Any idea how I can get around this issue? Feel free to move this issue to vitejs/vite when it is not directly related to nuxt.

### Additional context

Probably this error occurs because there is a dynamic import statement in `verovio/wasm-hum` (which is created by Emscripten):

```js
const{createRequire:createRequire}=await import("module");
```

### Logs

```shell-script
[plugin:vite:asset-import-meta-url] Failed to resolve entry for package "~/node_modules/.cache/vite/client/deps". The package may have incorrect main/module/exports specified in its package.json.
~/node_modules/.cache/vite/client/deps/verovio_wasm-hum.js
```

**Comment by WolfgangDrescher** [comment]
Yes, it is the same issue without Nuxt: https://stackblitz.com/edit/github-rjaduf?file=src%2Fpages%2FHome.vue

Change Vite to `~4.2.0` in `package.json` if you want to see what the output should actually look like.

@danielroe should I report this as a new issue in the Vite repo or can you move it over?

**Comment by WolfgangDrescher** [comment]
Probably not related to this issue, but when I lock Nuxt to `3.4.1` all CSS is missing on the website when I run `npm run build && node .output/server/index.mjs`. I do not have a reproduction for this other than my project: https://github.com/WolfgangDrescher/lassus-tricinium-project/tree/a9c3409b4d491d6fff2fc7400d95d9452adee94f

**Comment by WolfgangDrescher** [comment]
So far I have not received any feedback on the issue in the Vite repository. But since the Nuxt team surely has a good knowledge about Vite I would like to ask here again how I could address and debug this issue. Does anyone have any tips for me where would be a good starting point to fix the issue?

**Comment by danielroe** [comment]
This seems to work in the latest version of Nuxt/Vite.

**Comment by WolfgangDrescher** [comment]
Thank you for the update. It looks like it is working again since Vite `v5.1.1`. At least for my use case:

https://stackblitz.com/edit/github-rjaduf-caswxa?file=package.json
